src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenByUsernameAndPasswordConfidentialParameterBuilder.cs

```diff
@@ -67,5 +67,16 @@ internal override ApiEvent.ApiIds CalculateApiEventId()
         {
             return ApiEvent.ApiIds.AcquireTokenByUsernamePassword;
         }
+
+        /// <inheritdoc/>
+        protected override void Validate()
+        {
+            base.Validate();
+
+            if (Parameters.SendX5C == null)
+            {
+                Parameters.SendX5C = ServiceBundle.Config?.SendX5C ?? false;
```

Comment:
Why wouldnt this be set globally when it is enabled on the `WithCertificate(...)` method?

Also, should we add `WithSendX5C()` to this class?

Reply:
As for `Why wouldnt this be set globally when it is enabled on the WithCertificate(...) method?` question. It would require quite a big refactoring to do this and I'm not sure it's worth it, as we don't add new flows except very rarely. It would also introduce (small) breaking changes, which I don't think it's worth risking.

- SendX5C is defined on `AbstractAcquireTokenConfidentialClientParameters`, but ROPCParameters doesn't inherit this
- `AbstractConfidentialClientAcquireTokenParameterBuilder` doesn't have access to  `AbstractAcquireTokenConfidentialClientParameters` to set this in Validate()